src/debug_utils.cc

```diff
@@ -100,35 +100,104 @@ class Win32SymbolDebuggingContext final : public NativeSymbolDebuggingContext {
     USE(SymInitialize(current_process_, nullptr, true));
   }
 
-  ~Win32SymbolDebuggingContext() {
+  ~Win32SymbolDebuggingContext() override {
     USE(SymCleanup(current_process_));
   }
 
-  SymbolInfo LookupSymbol(void* address) override {
-    // Ref: https://msdn.microsoft.com/en-en/library/windows/desktop/ms680578(v=vs.85).aspx
-    char info_buf[sizeof(SYMBOL_INFO) + MAX_SYM_NAME];
-    SYMBOL_INFO* info = reinterpret_cast<SYMBOL_INFO*>(info_buf);
-    char demangled[MAX_SYM_NAME];
+  using NameAndDisplacement = std::pair<std::string, DWORD64>;
+  NameAndDisplacement WrappedSymFromAddr(DWORD64 dwAddress) const {
+    // Refs: https://docs.microsoft.com/en-us/windows/desktop/Debug/retrieving-symbol-information-by-address
+    // Patches:
+    // Use `fprintf(stderr, ` instead of `printf`
+    // `sym.filename = pSymbol->Name` on success
+    // `current_process_` instead of `hProcess.
+    DWORD64 dwDisplacement = 0;
+    // Patch: made into arg - DWORD64  dwAddress = SOME_ADDRESS;
+
+    char buffer[sizeof(SYMBOL_INFO) + MAX_SYM_NAME * sizeof(TCHAR)];
+    const auto pSymbol = reinterpret_cast<PSYMBOL_INFO>(buffer);
+
+    pSymbol->SizeOfStruct = sizeof(SYMBOL_INFO);
+    pSymbol->MaxNameLen = MAX_SYM_NAME;
+
+    if (SymFromAddr(current_process_, dwAddress, &dwDisplacement, pSymbol)) {
+      // SymFromAddr returned success
+      return NameAndDisplacement(pSymbol->Name, dwDisplacement);
+    } else {
+      // SymFromAddr failed
+      const DWORD error = GetLastError();  // "eat" the error anyway
+#ifdef DEBUG
+      fprintf(stderr, "SymFromAddr returned error : %lu\n", error);
+#endif
+    }
+    // End MSDN code
 
-    info->MaxNameLen = MAX_SYM_NAME;
-    info->SizeOfStruct = sizeof(SYMBOL_INFO);
+    return NameAndDisplacement();
+  }
 
-    SymbolInfo ret;
-    const bool have_info = SymFromAddr(current_process_,
-                                       reinterpret_cast<DWORD64>(address),
-                                       nullptr,
-                                       info);
-    if (have_info && strlen(info->Name) == 0) {
-      if (UnDecorateSymbolName(info->Name,
-                               demangled,
-                               sizeof(demangled),
-                               UNDNAME_COMPLETE)) {
-        ret.name = demangled;
-      } else {
-        ret.name = info->Name;
-      }
+  SymbolInfo WrappedGetLine(DWORD64 dwAddress) const {
+    SymbolInfo sym{};
+
+    // Refs: https://docs.microsoft.com/en-us/windows/desktop/Debug/retrieving-symbol-information-by-address
+    // Patches:
+    // Use `fprintf(stderr, ` instead of `printf`.
+    // Assign values to `sym` on success.
+    // `current_process_` instead of `hProcess.
```

Comment:
I am still not sure whether this is going to help us at some point, I suspect it won't.

Reply:
Comment are free.
Personally I have not been in a situation where I said "I wish there was less comments"

Reply:
I tend to agree, but someone will either change our code or the original, and updating the comment will mostly be a maintenance burden at that point from my perspective.